doc/7/getting-started/.react-native/screens/Chat/ChatClient.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react';
+import ChatView from './ChatView';
+import kuzzle from '../../services/kuzzle';
+
+class ChatClient extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            messages: []
+        };
+        this.kuzzle = kuzzle;
+        this.initConnection();
+        this.handleSendMessage = this.onSendMessage.bind(this);
+    }
+
+    async initConnection() {
+        // Etablish the connection
+        await kuzzle.connect();
+        // Check if 'chat' index exists
+        if (! await kuzzle.index.exists("chat")) {
+            // If not, create 'chat' index and 'messages' collection
+            await kuzzle.index.create("chat");
+            await kuzzle.collection.create("chat", "messages");
+        }
+        await this.fetchMessages();
+        await this.subscribeMessages();
+    }
+    displayDate(previousDate, currentDate) { 
+        if (previousDate === null) {// Message is the first of the array so need to display the date
+            return true;
+        }
+        const d1 = new Date(previousDate).toDateString()
+        const d2 = new Date(currentDate).toDateString()
+        if (d1 !== d2) {
+             // Previous message and current has different dates so need to display the date
+             return true
+        };
+        // Previous message and current has same dates so doesn't need to display the date
+        return false;
+    }
+    getMessage(hit, idx, array) {
+        // display will be set to true only if the previous message is from another day in goal to display only one time the dates
+        // and only the hours on each messages
+        let display = null;
+        // Not idx and array provided -> call from subscribe
+        let length = this.state.messages.length;
+        if (length == 0) {
+            display = false;
+        } else if (idx === null || array === null) {
+            display = this.displayDate( this.state.messages[length - 1].date, hit._source._kuzzle_info.createdAt)
+        } else { // idx and array provided -> call from fetch 
+            display = idx === 0? true : this.displayDate(array[idx - 1]._source._kuzzle_info.createdAt , hit._source._kuzzle_info.createdAt);
+        }
+        const message = {
```

Comment:
For better clarity, you should create this object at the beginning (with `displayDate` to `0`) and then use this object in the method